packages/cli/cloud/src/deploy-project/action.ts

```diff
@@ -0,0 +1,157 @@
+import fs from 'fs';
+import path from 'path';
+import { AxiosError } from 'axios';
+import * as crypto from 'node:crypto';
+import { apiConfig } from '../config/api';
+import { compressFilesToTar } from '../utils/compress-files';
+import createProjectAction from '../create-project/action';
+import type { CLIContext, ProjectInfos } from '../types';
+import { getTmpStoragePath } from '../config/local';
+import { cloudApiFactory, tokenServiceFactory, local } from '../services';
+import { notificationServiceFactory } from '../services/notification';
+import { loadPkg } from '../utils/pkg';
+import { buildLogsServiceFactory } from '../services/build-logs';
+
+const FILE_SIZE_LIMIT = 100 * 1024 * 1024; // 100MB
+
+type PackageJson = {
+  name: string;
+  strapi?: {
+    uuid: string;
+  };
+};
+
+async function upload(ctx: CLIContext, project: ProjectInfos, token: string) {
+  const cloudApi = cloudApiFactory(token);
+  // * Upload project
+  try {
+    const storagePath = getTmpStoragePath();
+    const projectFolder = path.resolve(process.cwd());
+    const packageJson = (await loadPkg(ctx)) as PackageJson;
+
+    if (!packageJson) {
+      ctx.logger.error(
+        'Unable to deploy the project. Please make sure the package.json file is correctly formatted.'
+      );
+      return;
+    }
+
+    ctx.logger.log('📦 Compressing project...');
+    // hash packageJson.name to avoid conflicts
+    const hashname = crypto.createHash('sha512').update(packageJson.name).digest('hex');
+    const compressedFilename = `${hashname}.tar.gz`;
+    try {
+      ctx.logger.debug(
+        'Compression parameters\n',
+        `Storage path: ${storagePath}\n`,
+        `Project folder: ${projectFolder}\n`,
+        `Compressed filename: ${compressedFilename}`
+      );
+      await compressFilesToTar(storagePath, projectFolder, compressedFilename);
+      ctx.logger.log('📦 Project compressed successfully!');
+    } catch (error: unknown) {
+      ctx.logger.error(
+        '⚠️ Project compression failed. Try again later or check for large/incompatible files.'
```

Comment:
why would trying again later do anything different here?

Reply:
It can be a transient issue that relates to their operating system.
We do not know where the CLI will be used, so we tried to put as many safe-guards as possible.